registry/default/magicui/animated-subscribe-button.tsx

```diff
@@ -1,66 +1,88 @@
 "use client";
 
+import { HTMLMotionProps } from "motion/react";
 import { AnimatePresence, motion } from "motion/react";
 import React, { useState } from "react";
 
-interface AnimatedSubscribeButtonProps {
+interface AnimatedSubscribeButtonProps extends HTMLMotionProps<"button"> {
   buttonColor: string;
   buttonTextColor?: string;
   subscribeStatus: boolean;
   initialText: React.ReactElement | string;
   changeText: React.ReactElement | string;
+  ref?: React.Ref<HTMLButtonElement>;
```

Comment:
this won't work since it overrides the forwardref

example:
```
"use client";
import { PulsatingButton } from "@/registry/default/magicui/pulsating-button";
import { useRef } from "react";

export default function PulsatingButtonDemo() {
  const ref = useRef<HTMLButtonElement>(null);

  const handleClick = () => {
    if (ref.current) {
      ref.current.style.backgroundColor = "red";
    }
  };
  return (
    <PulsatingButton ref={ref} onClick={handleClick}>
      Join Affiliate Program
    </PulsatingButton>
  );
}

```